Feature/conditionally disabled test execution button

jjelschen opened the pull request.

Commit:
dis-/enable run button based on selected file type

Commit:
update demo app module wrt workspace navigator / test execution service

Commit:
add test execution service mock for demo app

gunther-bachmann commented on src/lib/src/component/navigation/navigation.component.spec.ts:
```diff
@@ -33,13 +33,20 @@ describe('NavigationComponent', () => {
   let spy: jasmine.Spy;
   let sidenav: DebugElement;
 
-  let listedFile: WorkspaceElement = {
```
Just for the future, renaming would be better suited for a separate pull. Makes reviewing much easier

gunther-bachmann commented on src/lib/src/component/navigation/navigation.component.css:
```diff
@@ -54,6 +54,16 @@
   color: #ffffff;
 }
+
+:disabled {
```
Is this for disabled elements? Is that wanted?

franzbecker replied:
Shouldn't it be `.icon:disabled` ?

franzbecker commented on src/lib/src/component/navigation/navigation.component.spec.ts:
```diff
@@ -364,18 +371,61 @@ describe('NavigationComponent', () => {
+      type: ElementType.Folder,
+      children: [ nonExecutableFile, tclFile ]
+    });
+    fixture.detectChanges();
```
This code and the one above is redundant. We should probably extract this into a separate method.

franzbecker commented on src/lib/src/component/navigation/navigation.component.spec.ts:
```diff
@@ -364,18 +371,61 @@ describe('NavigationComponent', () => {
+    });
+    fixture.detectChanges();
+    let runIcon = sidenav.query(By.css('#run'));
+    component.uiState.selectedElement = nonExecutableFile;
```
I think this line is not doing anything. But I'm missing a test-case which is that the run button is disabled by default.

gunther-bachmann committed:
rework based on review

- css restricted to icon
- pointer off on disabled element
- hover texts for buttons
- extraction of test setup into separate file

franzbecker commented on src/lib/src/component/navigation/navigation.component.test.setup.ts:
```diff
@@ -0,0 +1,66 @@
+}
+
+export function mockedTestExecutionService() {
+  // Mock TestExecutionService
```
redundant comment

franzbecker commented on src/lib/src/component/navigation/navigation.component.test.setup.ts:
```diff
@@ -0,0 +1,66 @@
+};
+
+export function mockedPersistenceService() {
+  // Mock PersistenceService
```
redundant comment

gunther-bachmann committed:
correct test execution url usage

gunther-bachmann committed:
remove superfluous comments
